### ISSUE TITLE
Use proper pythonic exceptions

### ISSUE THREAD

**Issue by vsajip** [vsajip/python-gnupg#62]
**[Original report](https://bitbucket.org/vinay.sajip/python-gnupg/issue/62) by Kristoffer Berdal (Bitbucket: [flexd](https://bitbucket.org/flexd), GitHub: [flexd](https://github.com/flexd)).**

----------------------------------------

Currently when you encrypt or decrypt data (and probably all the other actions), you have to check result.status to see if the operation was successful.

It would make more sense to use normal [python exceptions](https://docs.python.org/2.7/tutorial/errors.html) so that we could catch the errors and handle them. 

I could make a pull-request when I have the time if you want. It would mean major interface changes to the whole library. Probably not until after christmas :/ It should probably be discussed a bit, but this is how you normally handle things in Python, and is probably a reason why someone would want to avoid using the library.

Basically changing all of this https://bitbucket.org/vinay.sajip/python-gnupg/src/22bcd34bf40cbb9994389688ccfbf52f8f776202/gnupg.py?at=default&fileviewer=file-view-default#gnupg.py-239 into actual Exceptions that can be handled when calling one of these functions.
